lib/encode_url.js

```diff
@@ -0,0 +1,17 @@
+'use strict';
+
+const { parse, format } = require('url');
+
+function encodeURL(str) {
+  if (parse(str).protocol) {
+    return format({
+      protocol: parse(str).protocol,
+      hostname: parse(str).hostname,
+      pathname: encodeURI(decodeURI(parse(str).pathname))
```

Comment:
decode+encode is necessary to avoid double-encode the space character; the space character is  encoded by url api by default.